helpers/order/order.go

```diff
@@ -38,11 +38,14 @@ func CheckBillingOrderStatus(sess *session.Session, receipt *datatypes.Container
 		return false, nil, err
 	}
 
-	currentStatus := *item.BillingItem.ProvisionTransaction.TransactionStatus.Name
-	for _, status := range statuses {
-		if currentStatus == status {
-			return true, &item, nil
+	if item.BillingItem != nil {
+		currentStatus := *item.BillingItem.ProvisionTransaction.TransactionStatus.Name
+		for _, status := range statuses {
+			if currentStatus == status {
+				return true, &item, nil
+			}
```

Comment:
Use the sl.GrabOk helper. https://github.com/softlayer/softlayer-go/blob/master/sl/helpers.go#L143

```go
currentStatus, ok := sl.GrabOk(item, "BillingItem.ProvisionTransaction.TransactionStatus.Name")
if ok {
    ...
}
```
Otherwise you would have to check every element in the tree for nil to make absolutely sure. This is why we created the helper.